typescript/cli/src/utils/input.ts

```diff
@@ -53,3 +71,393 @@ export async function inputWithInfo({
+ */
+
+type Status = 'loading' | 'idle' | 'done';
+
+type SearchableCheckboxTheme = {
+  icon: {
+    checked: string;
+    unchecked: string;
+    cursor: string;
+  };
+  style: {
+    disabledChoice: (text: string) => string;
+    renderSelectedChoices: <T>(
+      selectedChoices: ReadonlyArray<NormalizedChoice<T>>,
+      allChoices: ReadonlyArray<NormalizedChoice<T> | Separator>,
+    ) => string;
+    description: (text: string) => string;
+    helpTip: (text: string) => string;
+  };
+  helpMode: 'always' | 'never' | 'auto';
+};
+
+const checkboxTheme: SearchableCheckboxTheme = {
+  icon: {
+    checked: chalk.green(figures.circleFilled),
+    unchecked: figures.circle,
+    cursor: figures.pointer,
+  },
+  style: {
+    disabledChoice: (text: string) => chalk.dim(`- ${text}`),
+    renderSelectedChoices: (selectedChoices) =>
+      selectedChoices.map((choice) => choice.short).join(', '),
+    description: (text: string) => chalk.cyan(text),
+    helpTip: (text) => ` ${text}`,
+  },
+  helpMode: 'always',
+};
+
+export type SearchableCheckboxChoice<Value> = {
+  value: Value;
+  name?: string;
+  description?: string;
+  short?: string;
+  disabled?: boolean | string;
+  checked?: boolean;
+};
+
+type NormalizedChoice<Value> = Required<
+  Omit<SearchableCheckboxChoice<Value>, 'description'>
+> & {
+  description?: string;
+};
+
+type SearchableCheckboxConfig<Value> = {
+  message: string;
+  prefix?: string;
+  pageSize?: number;
+  instructions?: string;
+  choices: ReadonlyArray<SearchableCheckboxChoice<Value>>;
+  loop?: boolean;
+  required?: boolean;
+  selectableOptionsSeparator?: Separator;
+  validate?: (
+    choices: ReadonlyArray<SearchableCheckboxChoice<Value>>,
+  ) => boolean | string | Promise<string | boolean>;
+  theme?: PartialDeep<Theme<SearchableCheckboxTheme>>;
+};
+
+type Item<Value> = NormalizedChoice<Value> | Separator;
+
+type SearchableCheckboxState<Value> = {
+  options: Item<Value>[];
+  currentOptionState: Record<string, NormalizedChoice<Value>>;
+};
+
+function isSelectable<Value>(
+  item: Item<Value>,
+): item is NormalizedChoice<Value> {
+  return !Separator.isSeparator(item) && !item.disabled;
+}
+
+function isChecked<Value>(item: Item<Value>): item is NormalizedChoice<Value> {
+  return isSelectable(item) && Boolean(item.checked);
+}
+
+function toggle<Value>(item: Item<Value>): Item<Value> {
+  return isSelectable(item) ? { ...item, checked: !item.checked } : item;
+}
+
+function normalizeChoices<Value>(
+  choices: ReadonlyArray<SearchableCheckboxChoice<Value>>,
+): NormalizedChoice<Value>[] {
+  return choices.map((choice) => {
+    const name = choice.name ?? String(choice.value);
+    return {
+      value: choice.value,
+      name,
+      short: choice.short ?? name,
+      description: choice.description,
+      disabled: choice.disabled ?? false,
+      checked: choice.checked ?? false,
+    };
+  });
+}
+
+function sortNormalizedItems<Value>(
```

Comment:
FYI, localeCompare may be of use here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare#sort_an_array